internal/config/config.go

```diff
@@ -670,23 +677,33 @@ func bgpAdvertisementsFromLegacyCR(ads []metallbv1beta1.LegacyBgpAdvertisement,
 		ad.LocalPref = crdAd.LocalPref
 
 		for _, c := range crdAd.Communities {
-			if v, ok := communities[c]; ok {
-				ad.Communities[v] = true
-			} else {
-				v, err := ParseCommunity(c)
-				if err != nil {
-					return nil, fmt.Errorf("invalid community %q in BGP advertisement: %s", c, err)
-				}
-				ad.Communities[v] = true
+			v, err := getCommunityValue(c, communities)
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid community %q in BGP advertisement", c)
 			}
+			ad.Communities[v] = true
 		}
-
 		ret = append(ret, ad)
 	}
 
 	return ret, nil
 }
 
+func getCommunityValue(community string, communities map[string]uint32) (uint32, error) {
+	if v, ok := communities[community]; ok {
+		return v, nil
+	}
+
+	v, err := ParseCommunity(community)
+	if errors.Is(err, ErrFormat) {
+		return 0, err
+	} else if errors.Is(err, ErrInvalid) {
```

Comment:
nit: no need for the else, just another if

Reply:
done